android/pytorch_android_torchvision/src/main/cpp/pytorch_vision_jni.cpp

```diff
@@ -91,32 +95,64 @@ static void imageYUV420CenterCropToFloatBuffer(
   int xBeforeRtn, yBeforeRtn;
   int yi, yIdx, uvIdx, ui, vi, a0, ri, gi, bi;
   int channelSize = tensorWidth * tensorHeight;
-  int wr = outOffset;
-  int wg = wr + channelSize;
-  int wb = wg + channelSize;
-  for (int x = 0; x < tensorWidth; x++) {
+  // A bit of code duplication to avoid branching in the cycles
+  if (memoryFormatCode == kMemoryFormatContiguous) {
+    int wr = outOffset;
+    int wg = wr + channelSize;
+    int wb = wg + channelSize;
+    for (int y = 0; y < tensorHeight; y++) {
+      for (int x = 0; x < tensorWidth; x++) {
+        xBeforeRtn = cropXAdd + cropXMult * (int)(x * scale);
+        yBeforeRtn = cropYAdd + cropYMult * (int)(y * scale);
+        yIdx = yBeforeRtn * yRowStride + xBeforeRtn * yPixelStride;
+        uvIdx = (yBeforeRtn >> 1) * uvRowStride + (xBeforeRtn >> 1) * uvPixelStride;
+        ui = uData[uvIdx];
+        vi = vData[uvIdx];
+        yi = yData[yIdx];
+        yi = (yi - 16) < 0 ? 0 : (yi - 16);
+        ui -= 128;
+        vi -= 128;
+        a0 = 1192 * yi;
```

Comment:
what're these magic numbers for?

Reply:
This is yuv420 to rgb conversion https://en.wikipedia.org/wiki/YUV